toolchain/lowering/lowering_handle_expression_category.cpp

```diff
@@ -30,11 +30,18 @@ auto HandleBindValue(FunctionContext& context, SemIR::NodeId node_id,
   }
 }
 
-auto HandleMaterializeTemporary(FunctionContext& context, SemIR::NodeId node_id,
-                                SemIR::Node node) -> void {
+auto HandleTemporaryStorage(FunctionContext& context, SemIR::NodeId node_id,
+                            SemIR::Node node) -> void {
   context.SetLocal(
       node_id, context.builder().CreateAlloca(context.GetType(node.type_id()),
                                               nullptr, "temp"));
 }
 
+auto HandleTemporary(FunctionContext& context, SemIR::NodeId node_id,
+                     SemIR::Node node) -> void {
+  auto [temporary_id, init_id] = node.GetAsTemporary();
+  context.FinishInitialization(node.type_id(), temporary_id, init_id);
+  context.SetLocal(node_id, context.GetLocal(temporary_id));
+}
+
```

Comment:
```suggestion
auto HandleTemporary(FunctionContext& context, SemIR::NodeId node_id,
                     SemIR::Node node) -> void {
  auto [temporary_id, init_id] = node.GetAsTemporary();
  context.FinishInitialization(node.type_id(), temporary_id, init_id);
  context.SetLocal(node_id, context.GetLocal(temporary_id));
}

auto HandleTemporaryStorage(FunctionContext& context, SemIR::NodeId node_id,
                            SemIR::Node node) -> void {
  context.SetLocal(
      node_id, context.builder().CreateAlloca(context.GetType(node.type_id()),
                                              nullptr, "temp"));
}

```

Sorry, I'm nitpicking ordering. :-P